azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADAuthenticationFilterAutoConfiguration.java

```diff
@@ -26,6 +28,7 @@
 @ConditionalOnWebApplication
 @ConditionalOnProperty(prefix = "azure.activedirectory", value = {"client-id", "client-secret"})
 @EnableConfigurationProperties({AADAuthenticationProperties.class, ServiceEndpointsProperties.class})
+@PropertySource(value = "classpath:serviceEndpoints.properties")
 public class AADAuthenticationFilterAutoConfiguration {
```

Comment:
Is there a case that a user would load the `AADOAuth2AutoConfiguration` without the `AADAuthenticationFilterAutoConfiguration`? 

Looking at the code it would appear `AADAuthenticationFilterAutoConfiguration` depends on the `AADOAuth2AutoConfiguration` by proxy of the fact the `client-id` and `secret` properties are required for the UserService to function.

Reply:
They are two different auto configuration for different use cases. `AADOAuth2AutoConfiguration` mainly provides authorization mapping between AAD groups and Spring `GrantedAuthority`, the authorization leverages Spring Security OAuth2 native implementation.

While the `AADAuthenticationFilterAutoConfiguration` provides one way for user to provide ID_TOKEN from front end. 

The two samples [azure-active-directory-spring-boot-backend-sample](https://github.com/Microsoft/azure-spring-boot/blob/master/azure-spring-boot-samples/azure-active-directory-spring-boot-backend-sample/src/main/resources/application.properties) and [azure-active-directory-spring-boot-sample](https://github.com/Microsoft/azure-spring-boot/blob/master/azure-spring-boot-samples/azure-active-directory-spring-boot-sample/src/main/resources/application.properties) shows these 2 use cases.

Reply:
Updated the configuration to include the property source and test it.

Reply:
Thanks very much for the contribution, could you share what's the problem you have met by customizing `serviceEndpoints.yml`? It should work if the user configure own `serviceEndpoints.yml` under src/main/resources.